### PR TITLE
Remove change password page

### PR DESCRIPTION
**Pull request by rebecca-thompson**
## What does this change?
This redirects the `/password/change` page to the `/reset` page which is the new reset password page in Gateway and deletes unused code related to this endpoint. We are doing this to reduce the number of places where a user can reset their password to minimise maintenance and ensure security best practice.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Comment by rebecca-thompson** [comment]
[Research: There are endpoints for password reset, verify and onboarding that have not been migrated to gateway](https://trello.com/c/rHLxqsg9/2660-research-there-are-endpoints-for-password-reset-verify-and-onboarding-that-have-not-been-migrated-to-gateway)

**Comment by QuarpT** [comment]
looks great.

it would be good to make remove the migrated routes in this service entirely (by making changes to the clients that call the service or adding routing rules to Fastly). We'll have to do this before deleting the service.

**Review by QuarpT — APPROVED** [review]

**Comment by rebecca-thompson** [comment]
> looks great.
> 
> it would be good to remove the migrated routes in this service entirely (by making changes to the clients that call the service or adding routing rules to Fastly). We'll have to do this before deleting the service.

I was thinking this as well. I'll prepare a future pr to move all redirects to Fastly and remove the remaining code

File: identity/app/idapiclient/IdApiClient.scala [hunk 1]
```diff
@@ -85,48 +84,14 @@ class IdApiClient(idJsonBodyParser: IdApiJsonBodyParser, conf: IdConfig, httpCli
...
-  // PASSWORD RESET/UPDATE
-
-  def passwordExists(auth: Auth, trackingData: TrackingData): Future[Response[Boolean]] = {
-    val apiPath = urlJoin("user", "password-exists")
-    val headers = buildHeaders(Some(auth), extra = xForwardedForHeader(trackingData))
-    val response = httpClient.GET(apiUrl(apiPath), None, buildParams(Some(auth)), headers)
-    response map extract[Boolean](jsonField("passwordExists"))
-  }
-
-  def updatePassword(
```
[the remark this answers is not in the document]
Comment (rebecca-thompson) [reply identity/app/idapiclient/IdApiClient.scala:97]: Yes, I think so. I will check the access logs and remove if it if possible

**Pull request merged**